### PR TITLE
Align python crossfire parser to c equivalent

### PR DESCRIPTION
**Pull request by 3djc**
Since I do not have access to crossfire specs, I have updated the python scripts based on OpenTx 'c' crossfire parser.

The changes include support for negative values and GPS packets parsing

### CODE REVIEW

**Comment by 3djc** [comment]
This was tested based on logs submitted by tgreer, so it would be nice to test again different logs too, just to be sure

File: radio/util/crossfire-parse.py [hunk 1]
```diff
@@ -70,28 +71,47 @@ def dump(data, maxLen=None):
...
     crc = 0
     for c in buffer:
         crc = crc8tab[crc ^ c]
     return crc
 
+# note : python arrays slicing is a bit strange payload[x:y] means from 'x' included to 'y' NOT included
+#        therefore y - x gives the number of bytes sliced.
 def ParseGPS(payload):
-    pass
+    lat = getCrossfireValue(True, payload[0:4]) / 1e7
```
Comment (bsongis) [inline radio/util/crossfire-parse.py]: do you mean 0x1e7?

File: radio/util/crossfire-parse.py [hunk 2]
```diff
@@ -70,28 +71,47 @@ def dump(data, maxLen=None):
...
+    else:
+        return value.uint
+
 def crc8(buffer):
     crc = 0
     for c in buffer:
         crc = crc8tab[crc ^ c]
     return crc
 
+# note : python arrays slicing is a bit strange payload[x:y] means from 'x' included to 'y' NOT included
```
Comment (bsongis) [inline radio/util/crossfire-parse.py]: why is it strange? I prefer you remove this comment ;)

File: radio/util/crossfire-parse.py [hunk 3]
```diff
@@ -70,28 +71,47 @@ def dump(data, maxLen=None):
     0xAD, 0x78, 0xD2, 0x07, 0x53, 0x86, 0x2C, 0xF9
 ]
 
+# logs data are BigEndian
+def getCrossfireValue(signed, data=[], *args):
```
Comment (bsongis) [inline radio/util/crossfire-parse.py]: I have to find a nicer way to do this tomorrow :)

File: radio/util/crossfire-parse.py (hunk 1 above)
Comment (3djc) [reply radio/util/crossfire-parse.py]: Not quite, I mean 10 000 000

Python 2.7.13 (default, Jan 19 2017, 14:48:08) 
[GCC 6.3.0 20170118] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> print 1e7
10000000.0
>>>

File: radio/util/crossfire-parse.py (hunk 2 above)
Comment (3djc) [reply radio/util/crossfire-parse.py]: Well, it seems a lot of peop;le (including myself) bump into this all the time with python slicing, but I'll make it politicaly correct :)

File: radio/util/crossfire-parse.py (hunk 3 above)
Comment (3djc) [reply radio/util/crossfire-parse.py]: Well, you could easily remove the signed argument and assume that 4 bytes data are signed and less then 4 is unsigned, which seems to be true in current version, but as this is more like a tool for debuging, I wanted it to be crystal clear (unless thats not the 'signed' bothering you ?)

File: radio/util/crossfire-parse.py (hunk 1 above)
Comment (mpaperno) [reply radio/util/crossfire-parse.py]: Much easier to read than counting zeros, IMHO.  And another reason hex should be written with uppercase chars...  :)

**Comment by bsongis** [comment]
Thanks!

**Pull request merged**